satpy/etc/readers/msi_safe_l2a.yaml

```diff
@@ -0,0 +1,315 @@
+reader:
+  name: msi_safe_l2a
+  short_name: MSI SAFE L2A
+  long_name: Sentinel-2 A and B MSI L2A data in SAFE format
+  description: SAFE Reader for MSI L2A data (Sentinel-2)
+  status: Nominal
+  supports_fsspec: false
+  sensors: [msi]
+  default_channels: []
+  reader: !!python/name:satpy.readers.yaml_reader.FileYAMLReader
+
+file_types:
+  l2a_safe_granule_10m:
+        file_reader: !!python/name:satpy.readers.msi_safe.SAFEMSIL1C
+        file_patterns: ['{fmission_id:3s}_MSI{process_level:3s}_{observation_time:%Y%m%dT%H%M%S}_N{fprocessing_baseline_number:4d}_R{relative_orbit_number:3d}_T{dtile_number:5s}_{dproduct_discriminator:%Y%m%dT%H%M%S}.SAFE/GRANULE/L2A_T{gtile_number:5s}_A{absolute_orbit_number:6d}_{gfile_discriminator:%Y%m%dT%H%M%S}/IMG_DATA/R10m/T{tile_number:5s}_{file_discriminator:%Y%m%dT%H%M%S}_{band_name:3s}_10m.jp2']
+        requires: [l2a_safe_metadata, l2a_safe_tile_metadata]
+  l2a_safe_granule_20m:
+        file_reader: !!python/name:satpy.readers.msi_safe.SAFEMSIL1C
+        file_patterns: ['{fmission_id:3s}_MSI{process_level:3s}_{observation_time:%Y%m%dT%H%M%S}_N{fprocessing_baseline_number:4d}_R{relative_orbit_number:3d}_T{dtile_number:5s}_{dproduct_discriminator:%Y%m%dT%H%M%S}.SAFE/GRANULE/L2A_T{gtile_number:5s}_A{absolute_orbit_number:6d}_{gfile_discriminator:%Y%m%dT%H%M%S}/IMG_DATA/R20m/T{tile_number:5s}_{file_discriminator:%Y%m%dT%H%M%S}_{band_name:3s}_20m.jp2']
+        requires: [l2a_safe_metadata, l2a_safe_tile_metadata]
+  l2a_safe_granule_60m:
+        file_reader: !!python/name:satpy.readers.msi_safe.SAFEMSIL1C
+        file_patterns: ['{fmission_id:3s}_MSI{process_level:3s}_{observation_time:%Y%m%dT%H%M%S}_N{fprocessing_baseline_number:4d}_R{relative_orbit_number:3d}_T{dtile_number:5s}_{dproduct_discriminator:%Y%m%dT%H%M%S}.SAFE/GRANULE/L2A_T{gtile_number:5s}_A{absolute_orbit_number:6d}_{gfile_discriminator:%Y%m%dT%H%M%S}/IMG_DATA/R60m/T{tile_number:5s}_{file_discriminator:%Y%m%dT%H%M%S}_{band_name:3s}_60m.jp2']
+        requires: [l2a_safe_metadata, l2a_safe_tile_metadata]
+  l2a_safe_tile_metadata:
+        file_reader: !!python/name:satpy.readers.msi_safe.SAFEMSITileMDXML
+        file_patterns: ['{fmission_id:3s}_MSI{process_level:3s}_{observation_time:%Y%m%dT%H%M%S}_N{fprocessing_baseline_number:4d}_R{relative_orbit_number:3d}_T{dtile_number:5s}_{dproduct_discriminator:%Y%m%dT%H%M%S}.SAFE/GRANULE/L2A_T{gtile_number:5s}_A{absolute_orbit_number:6d}_{gfile_discriminator:%Y%m%dT%H%M%S}/MTD_TL.xml']
+  l2a_safe_metadata:
+        file_reader: !!python/name:satpy.readers.msi_safe.SAFEMSIMDXML
+        file_patterns: ['{fmission_id:3s}_MSI{process_level:3s}_{observation_time:%Y%m%dT%H%M%S}_N{fprocessing_baseline_number:4d}_R{relative_orbit_number:3d}_T{dtile_number:5s}_{dproduct_discriminator:%Y%m%dT%H%M%S}.SAFE/MTD_MSIL2A.xml']
+
+datasets:
+  B01_L2A:
+    name: B01_L2A
```

Comment:
So, what is the difference between B01 in l1c and B01 in l2a? can it be expressed with modifiers?
I'm thinking in the l1c yaml file, it would be like:
```yaml
B01:
  name: B01
  sensor: msi
 ...
```
and in l2a
```yaml
B01:
  name: B01
  sensor: msi
  modifiers: [fancy_sunz_correction, fancy_rayleigh_correction]
...
```

That way, you can use these modifiers to identify the right channel levels in the composites yaml.

Reply:
I'm not sure if we really need this because they're already in different file types. There're also different composites for l1c and l2a products.

Reply:
Ok, but do we really need to give them names ending with `L2A` ? We have different readers with the same band names and that is not a problem...